packages/compiler/src/module-resolver/module-resolver.ts

```diff
@@ -123,59 +153,128 @@ export async function resolveModule(
     return paths;
   }
 
-  function getPackageCandidates(
-    name: string,
-    baseDir: string,
-  ): Array<{ path: string; type: "node_modules" | "self" }> {
-    const dirs = listAllParentDirs(baseDir);
-    return dirs.flatMap((x) => [
-      { path: x, type: "self" },
-      { path: joinPaths(x, "node_modules", name), type: "node_modules" },
-    ]);
+  /**
+   * Equivalent implementation to node LOAD_PACKAGE_SELF
+   * Resolve if the import is importing the current package.
+   */
+  async function resolveSelf(name: string, baseDir: string): Promise<ResolvedModule | undefined> {
+    for (const dir of listDirHierarchy(baseDir)) {
+      const pkgFile = resolvePath(dir, "package.json");
+      if (!(await isFile(host, pkgFile))) continue;
+      const pkg = await readPackage(host, pkgFile);
+      if (pkg.name === name) {
+        return loadPackage(dir, pkg);
+      } else {
+        return undefined;
```

Comment:
@timotheeguerin here is a bug which caused not found module issue.

https://dev.azure.com/azure-sdk/public/_build/results?buildId=4218379&view=logs&j=aa4563e2-7f5a-5a88-019c-341451fd8a67&t=3351c2db-8046-5507-7e56-867b3895d138&l=38

```
error import-not-found: Couldn't resolve import "@azure-tools/typespec-ts"
```

Reply:
I think the way it is done now is the correct way according the the node resolution spec, you only lookup to the first parent package.json for resolving self. What is the use case?